app/src/org/commcare/android/javarosa/IntentCallout.java

```diff
@@ -303,10 +309,53 @@ private void storePointerToFileResponse(TreeReference ref, String responseValue,
         if (!src.exists()) {
             //TODO: How hard should we be failing here?
             Log.w(TAG, "CommCare received a link to a file at " + src.toString() + " to be included in the form, but it was not present on the phone!");
+
+            Uri uri = Uri.parse(responseValue);
+
+            try {
+                ParcelFileDescriptor inFile = androidContext.getContentResolver().openFileDescriptor(uri, "r");
+
+                File newFile = new File(destinationFile, uri.getLastPathSegment());
```

Comment:
I'm not sure whether `uri.getLastPathSegment()` always return the correct fileName. I think it sometimes returns a random number and that might create issues later(formSubmission task checks for file extensions before sending them).

Reply:
Updated this to retrieve the filename from the content provider directly if possible, and revert to the uri path segment only as a fall back, thanks for raising